File: packages/gamole_types/schemas/generated.py
"""Generated output schemas - ported from packages/types/src/schemas/generated.ts."""

from pydantic import BaseModel, Field


class GeneratedStory(BaseModel):
    title: str
    description: str
    acceptance_criteria: list[str] = Field(alias="acceptanceCriteria")
    assumptions: list[str]
    technical_notes: str | None = Field(default=None, alias="technicalNotes")

    model_config = {"populate_by_name": True}


class GeneratedEpic(BaseModel):
    epic_title: str = Field(alias="epicTitle")
    epic_description: str = Field(alias="epicDescription")
    team_name: str | None = Field(default=None, alias="teamName")
    team_reason: str | None = Field(default=None, alias="teamReason")
    stories: list[GeneratedStory]

    model_config = {"populate_by_name": True}


class GeneratedOutput(BaseModel):
    epics: list[GeneratedEpic]
    project_name: str | None = Field(default=None, alias="projectName")
    overall_notes: str | None = Field(default=None, alias="overallNotes")

    model_config = {"populate_by_name": True}

File: packages/types/src/schemas/generated.ts
import { z } from 'zod';

// GeneratedStory schema
export const GeneratedStorySchema = z.object({
  title: z.string(),
  description: z.string(),
  acceptanceCriteria: z.array(z.string()),
  assumptions: z.array(z.string()),
  technicalNotes: z.string().optional(),
});

export type GeneratedStory = z.infer<typeof GeneratedStorySchema>;

// GeneratedEpic schema
export const GeneratedEpicSchema = z.object({
  epicTitle: z.string(),
  epicDescription: z.string(),
  stories: z.array(GeneratedStorySchema),
});

export type GeneratedEpic = z.infer<typeof GeneratedEpicSchema>;

// GeneratedOutput schema
export const GeneratedOutputSchema = z.object({
  epics: z.array(GeneratedEpicSchema),
  overallNotes: z.string().optional(),
});

export type GeneratedOutput = z.infer<typeof GeneratedOutputSchema>;
